src/distance.jl

```diff
@@ -5,6 +5,11 @@ struct DefaultDistance<:AbstractMatrix{Int}
     DefaultDistance(nv::Int=typemax(Int)) = new(nv)
 end
 
+DefaultDistance(nv::Integer) = DefaultDistance(Int(nv))
```

Comment:
Could you please elaborate on this line @sbromberger? What integer types are we planning to support?